### PR TITLE
sp - implements webdav specific http verbs

### PR DESCRIPTION
**Pull request by pmauduit**
This is required in order to be able to proxify webdav requests, e.g. to interact with a NextCloud instance behind the SP.

### CODE REVIEW

File: security-proxy/src/main/java/org/georchestra/security/Proxy.java [hunk 1]
```diff
@@ -957,7 +958,25 @@ private HttpRequestBase makeRequest(HttpServletRequest request, String sURL) thr
...
+                pfm.setEntity(entity);
+                return pfm;
+            } else if (SearchMethod.METHOD_NAME.equalsIgnoreCase(method)) {
+                SearchMethod sm = new SearchMethod(uri);
+                int contentLength = request.getContentLength();
+                ServletInputStream inputStream = request.getInputStream();
+                HttpEntity entity = new InputStreamEntity(inputStream, contentLength);
+                sm.setEntity(entity);
+                return sm;
+            }
```
Comment (RemiDesgrange) [inline security-proxy/src/main/java/org/georchestra/security/Proxy.java:977]: I'm wondering in which case the two conditions couldn't be fullfiled?

Comment (pmauduit) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java:977]: if the coming request is a completely custom HTTP verb ? But then I'd expect an exception to be thrown here:
https://github.com/georchestra/georchestra/pull/3483/files#diff-557c2eb148c461ecfcde83d38082b5270605be793cd92a707660c2f38e527adfR1053

Comment (pmauduit) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java:977]: do you think it deserves an extra test ?

Comment (pmauduit) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java:977]: trying an extra test shows that `switch(null)` throws an npe.

Comment (pmauduit) [reply security-proxy/src/main/java/org/georchestra/security/Proxy.java:977]: I added a test & a guard to avoid the NPE.

**Comment by pmauduit** [comment]
superseded by #3528

**Comment by RemiDesgrange** [comment]
closed since #3569 superseed this.
